src/components/HOC/AddFavoriteHOC.jsx

```diff
@@ -1,17 +1,33 @@
 import React from "react";
 import { API_URL, API_KEY_3, fetchApi } from "../../api/api";
+import _ from "lodash";
+import { inject, observer } from "mobx-react";
 
 export default (Component, type) =>
+  @inject(({ userStore, loginFormStore }) => ({
+    userStore,
+    loginFormStore
+  }))
+  @observer
   class AddFavoriteHOC extends React.Component {
     constructor(props) {
       super(props);
       this.state = {
-        idAdded: props[type].includes(props.item.id)
+        isAdded: this.getAddById({
+          list: props.userStore[type],
+          id: props.item.id
+        })
       };
     }
 
+    getAddById = ({ list, id }) => {
+      list.some(item => item.id === id);
+    };
+
     onAddFavorites = name => () => {
-      const { user, session_id, item, toggleModal } = this.props;
+      const { item } = this.props;
+      const { user, session_id } = this.props.userStore;
```

Comment:
ты не умеешь глубокую деструктурузацию еще делать? :)
const { user, session_id } = this.props.userStore; =>
const { userStore: { user, session_id } } = this.props;
```suggestion
      const { user, session_id } = this.props.userStore;
```